api/include/opentelemetry/trace/span_context.h

```diff
@@ -81,6 +94,7 @@ class SpanContext final
   trace_api::TraceId trace_id_;
   trace_api::SpanId span_id_;
   trace_api::TraceFlags trace_flags_;
+  trace_api::TraceState trace_state_;
```

Comment:
I see some benefits of storing this by value. But I'm leaning a bit more toward a `nostd::shared_ptr<TraceState>` approach. I think we'll mostly have two scenarios:
1. No `TraceState` is used at all. In that instance it would be cheaper to have a `shared_ptr` that can be a `nullptr`.
2. The same `TraceState` is used for all spans in a trace. in that case, a `shared_ptr` could be shared for all spans.

I'm still not sure what's better. I'm just a bit concerned about adding ~ 500 bytes to every span (context).

Reply:
Share the state for all the Spans in the same traces is a good point. Changed the type to `nostd::shared_ptr`.